images/ipfailover/keepalived/lib/config-generators.sh

```diff
@@ -214,15 +182,27 @@ function generate_vrrpd_instance_config() {
 
   local vipname=$(scrub "$1")
   local initialstate=""
-  local preempt=${PREEMPTION:-"$DEFAULT_PREEMPTION_STRATEGY"}
+  local preempt=${PREEMPTION:-"${DEFAULT_PREEMPTION_STRATEGY}"}
   local vrrpidoffset=${HA_VRRP_ID_OFFSET:-0}
 
-  [ "$instancetype" = "master" ] && initialstate="state MASTER"
+  [ "${instancetype}" = "master" ] && initialstate="state MASTER"
 
-  local instance_name=$(generate_vrrp_instance_name "$servicename" "$iid")
+  local instance_name=$(generate_vrrp_instance_name "${servicename}" "${iid}")
```

Comment:
You should not mix scoping statements and command substitutions in one command, exit code aliasing will occur.

```sh
local instance_name; instance_name=$(generate_vrrp_instance_name "${servicename}" "${iid}")
```